```
> misskey-js@2024.5.0 api-prod /home/runner/work/misskey/misskey/packages/misskey-js
> pnpm api-extractor run --verbose


api-extractor 7.47.0  - https://api-extractor.com/

Using configuration from ./api-extractor.json
Analysis will use the bundled TypeScript version 5.4.2
*** The target project appears to use TypeScript 5.5.3 which is newer than the bundled compiler engine; consider upgrading API Extractor.
Writing: /home/runner/work/misskey/misskey/packages/misskey-js/temp/misskey-js.api.json
Generating complete API report: /home/runner/work/misskey/misskey/packages/misskey-js/etc/misskey-js.api.md
Warning: You have changed the API signature for this project. Please copy the file "temp/misskey-js.api.md" to "etc/misskey-js.api.md", or perform a local build (which does this automatically). See the Git repo documentation for more info.

API Extractor completed with warnings
/home/runner/work/misskey/misskey/packages/misskey-js:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  misskey-js@2024.5.0 api-prod: `pnpm api-extractor run --verbose`
Exit status 1
Error: Process completed with exit code 1.
```
謎

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -37,6 +37,7 @@
 
 ### Misskey.js
 - Feat: `/drive/files/create` のリクエストに対応（`multipart/form-data`に対応）
+- Feat: `/admin/role/create` のロールポリシーの型を修正
 
 ## 2024.5.0
 

--- a/packages/misskey-js/etc/misskey-js.api.md
+++ b/packages/misskey-js/etc/misskey-js.api.md
@@ -2736,7 +2736,7 @@ function parse(acct: string): Acct;
 //
 // @public (undocumented)
 type PartialRolePolicyOverride = Partial<{
-    [k in keyof RolePolicies]: Omit<Values<Role["policies"]>, "value"> & {
+    [k in keyof RolePolicies]: Omit<Values<Role['policies']>, 'value'> & {
         value: RolePolicies[k];
     };
 }>;

--- a/packages/misskey-js/test/api.ts
+++ b/packages/misskey-js/test/api.ts
@@ -259,4 +259,35 @@ describe('API', () => {
 			expect(isAPIError(e)).toEqual(false);
 		}
 	});
+
+	test('admin/roles/create の型が合う', async() => {
+		fetchMock.resetMocks();
+		// レスポンスに依存しないのでホックの注入は不要
+		const cli = new APIClient({
+			origin: 'https://misskey.test',
+			credential: 'TOKEN',
+		});
+		await cli.request('admin/roles/create', {
+			name: 'aaa',
+			asBadge: false,
+			canEditMembersByModerator: false,
+			color: "#123456",
+			condFormula: {},
+			description: '',
+			displayOrder: 0,
+			iconUrl: "",
+			isAdministrator: false,
+			isExplorable: false,
+			isModerator: false,
+			isPublic: false,
+			policies: {
+				ltlAvailable: {
+					value: true,
+					priority: 0,
+					useDefault: false,
+				},
+			},
+			target: "manual",
+		});
+	})
 });

